File: Python/lastfactorialdigit.py
'''
 *
 *	Problem: "Last Factorial Digit"
 *
'''

import sys

numCases = input()
numCases = int(numCases)

case = 1
while (case <= numCases):

	#print(case)
	
	n = input()
	n = int(n)
	
	factorial = 1
	index = 1
	while (index < n):
		factorial *= (index + 1)
		index += 1
		
	lastDigit = factorial % 10
	print(lastDigit)
	case += 1

File: Ruby/lastfactorialdigit.rb
=begin
 *
 *	Problem:		Last Factorial Digit
 *	Problem ID:	lastfactorialdigit
 *	url:				https://open.kattis.com/problems/lastfactorialdigit
 *	Solution:		Ruby
 *
=end

class Solver

	def solve
		number_cases = gets.chomp.to_i
		for this_case in 1..number_cases
			n = gets.chomp.to_i
			factorial = 1
			index = 1
			while (index < n)
				factorial *= (index + 1)
				index += 1
			end
			last_digit = factorial % 10
			puts(last_digit)
		end
	end
	
end

solver = Solver.new
solver.solve
